src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlDatabaseAuditingPolicy.cs

```diff
@@ -135,6 +144,10 @@ private void ApplyUserInputToBlobAuditingModel(DatabaseBlobAuditingPolicyModel m
                 model.AuditAction = AuditAction;
             }
 
+            if (EventType != null) // Event types are relevant only for Table auditing
+            {
+                throw new Exception(string.Format(Properties.Resources.EventTypeConfiguringIrrelevantForBlobAuditingPolicy));
```

Comment:
This would also help with all the other places where you are checking what parameters are set based on whether they are using table or blob auditing.